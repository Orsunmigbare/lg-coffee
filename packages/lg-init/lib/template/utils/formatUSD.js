/* eslint-disable import/prefer-default-export */
const formatUSD = (number) => {
  const string = JSON.stringify(number);
  const { length } = string;
  return JSON.parse(
    `${string.substring(0, length - 2)}.${string.substring(length - 2)}`
  ).toFixed(2);
};

export { formatUSD };
