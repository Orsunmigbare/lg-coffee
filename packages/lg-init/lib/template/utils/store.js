module.exports = {
  URL: process.env.LOCAL_URL || process.env.PUBLIC_URL,
  storeName: 'VisionSpeaks',
  logo: '/logo.png',
  accountOwner: 'Yulissa V',
  products: [
    {
      name: 'Lady In The Void',
      description: 'Holographic Sticker, This thing is bad ass',
      price: '287',
      sku: 'lad_vs_',
      images: [
        '/lady_in_the_void_TRANSP.png',
        '/lady_001.png',
        '/lady_002.png',
      ],
    },
    {
      name: 'Bunns In The Void',
      description: 'Holographic Sticker, This thing is bad ass',
      price: '287',
      sku: 'bun_vs_',
      images: [
        '/bunns_in_the_void_TRANSP.png',
        '/bunns_001.png',
        '/bunns_002.png',
      ],
    },
  ],
  pages: {
    globals: {
      // navigations
      navigation: [
        {
          title: 'Products',
          path: '/',
        },
        {
          title: 'Made To Order',
          path: '/info/contact',
        },
        {
          title: 'About',
          path: '/info/about',
        },
        {
          title: 'Cart',
          path: '/view-cart',
        },
      ],
      footer: [
        {
          title: 'Contact',
          path: 'mailto:support@visionspeaks.art',
        },
        {
          title: 'FAQ',
          path: '/info/faq',
        },
      ],
    },
    homePage: {
      hero: {
        image: '/heatpress.jpeg',
        headline: 'Made To Order',
        subHeadline: 'The highest quality work comes from custom work.',
        buttonName: 'Start Your Order',
        buttonPath: '/products',
        styles: {},
      },
    },
    info: {
      faq: [
        {
          q: 'Do you make custom products?',
          a: 'Yes',
        },
        {
          q: 'How long will it take to recieve my purchase?',
          a: 'Unless otherwise stated, All orders go out via Priority Mail.',
        },
        {
          q: 'What is your policy on refunds?',
          a: 'At this time, we are unable to honor refunds.',
        },
        {
          q:
            "I never received my purchase or something isn't right with my order. What can I do?",
          a: 'Reach out to support@visionspeaks.art.',
        },
      ],
      about: `
        VisionSpeaks is a custom art collective.
      `,
    },
  },
};
