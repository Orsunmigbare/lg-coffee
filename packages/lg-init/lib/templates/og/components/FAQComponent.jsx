/* eslint-disable react/require-default-props */
import React from 'react';
import PropTypes from 'prop-types';
import { H4, P } from './_helpers';

const FAQComponent = ({ store }) => (
  <div className="p-4">
    {store.info.faq.map((i) => (
      <div key={i.q}>
        <H4>{i.q}</H4>
        <P>{i.a}</P>
      </div>
    ))}
  </div>
);

FAQComponent.propTypes = {
  store: PropTypes.shape({
    info: PropTypes.shape({
      faq: PropTypes.arrayOf(PropTypes.shape({})),
    }),
  }),
};

export default FAQComponent;
