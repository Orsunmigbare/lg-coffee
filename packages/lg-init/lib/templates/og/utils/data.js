exports.data = {
  store: {
    name: 'VisionSpeaks',
    logo: '/logo.png',
    url: '',
    description: 'Custom sticker shop',
    contact: 'youremail@here.com',
    info: {
      faq: [
        {
          q: 'Do you make custom products?',
          a: 'Yes',
        },
        {
          q: 'How long will it take to recieve my purchase?',
          a: 'Unless otherwise stated, All orders go out via Priority Mail.',
        },
        {
          q: 'What is your policy on refunds?',
          a: 'At this time, we are unable to honor refunds.',
        },
        {
          q:
            "I never received my purchase or something isn't right with my order. What can I do?",
          a: 'Reach out to support@visionspeaks.art.',
        },
      ],
    },
    social: {
      facebook: '',
      twitter: '',
      instagram: '',
    },
  },
  theme: {
    text: { color: '#212529' },
    secondaryText: { color: '#6C757D' },
    primaryBackground: { backgroundColor: '#F8F9FA' },
    secondaryBackground: { backgroundColor: '#FFFFFF' },
  },
  products: {
    heading: 'New Arrivials',
    products: [
      {
        name: 'Bunns In The Void',
        description: 'Holographic Sticker, This thing is bad ass',
        price: 287,
        sku: 'bun_vs_',
        images: [
          '/bunns_in_the_void_TRANSP.png',
          '/bunns_001.png',
          '/bunns_002.png',
        ],
        type: 'sticker',
        qty: 40,
      },
      {
        name: 'Lady In The Void',
        description: 'Holographic Sticker, This thing is bad ass',
        price: 287,
        sku: 'lad_vs_',
        images: [
          '/lady_in_the_void_TRANSP.png',
          '/lady_001.png',
          '/lady_002.png',
        ],
        type: 'sticker',
        qty: 40,
      },
    ],
  },
};
