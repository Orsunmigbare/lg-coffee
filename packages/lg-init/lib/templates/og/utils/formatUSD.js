/* eslint-disable import/prefer-default-export */
const formatUSD = (number) => {     
  const { length } = number;
  const str = `${number.substring(0, length - 2)}.${number.substring(length - 2)}`;  
  return parseFloat(str).toFixed(2)
};

export { formatUSD };
