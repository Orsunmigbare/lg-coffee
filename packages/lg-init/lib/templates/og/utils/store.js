exports.store = {
  name: 'Dog-Gone Goods',
  logo: '/logo.png',
  hero: '/hero.png',
  url: '',
  description: 'A dog treat shop',
  about: 'Everyone loves a happy dog!',
  contact: 'youremail@here.com',
  social: {
    facebook: '',
    twitter: '',
    instagram: '',
  },
};
