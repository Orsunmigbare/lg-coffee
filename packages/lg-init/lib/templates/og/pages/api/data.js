import { data } from '../../utils/data';

export default function handler(req, res) {
  res.statusCode = 200;
  res.setHeader('Content-Type', 'application/json');
  return res.end(JSON.stringify(data));
}
