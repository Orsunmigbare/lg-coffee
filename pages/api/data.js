import { data } from "../../utils/data";

export default function handler(req, res) {
  res.statusCode = 200;
  res.setHeader("Content-Type", "application/json");

  // fake db call
  setTimeout(() => {
    res.end(JSON.stringify(data));
  }, 1000);
}
