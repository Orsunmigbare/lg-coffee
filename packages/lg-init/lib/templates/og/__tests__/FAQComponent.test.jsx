import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import FAQComponent from '../components/FAQComponent';
import { data } from '../utils/data';

it('FAQComponent components renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<FAQComponent store={data.store} theme={data.theme} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('FAQComponent component renders unchanged', () => {
  const tree = renderer.create(<FAQComponent store={data.store} theme={data.theme} />).toJSON();
  expect(tree).toMatchSnapshot();
});
