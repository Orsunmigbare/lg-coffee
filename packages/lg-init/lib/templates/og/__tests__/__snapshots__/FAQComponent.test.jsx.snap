// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQComponent component renders unchanged 1`] = `
<div
  className="p-4"
>
  <div>
    <h4
      className=""
      style={
        Object {
          "color": "#212529",
        }
      }
    >
      Are these treats tasty?
    </h4>
    <p
      className=""
      style={
        Object {
          "color": "#212529",
        }
      }
    >
      Yes you bet they are!
    </p>
  </div>
  <div>
    <h4
      className=""
      style={
        Object {
          "color": "#212529",
        }
      }
    >
      How long will it take to recieve my purchase?
    </h4>
    <p
      className=""
      style={
        Object {
          "color": "#212529",
        }
      }
    >
      Unless otherwise stated, All orders go out via Priority Mail.
    </p>
  </div>
  <div>
    <h4
      className=""
      style={
        Object {
          "color": "#212529",
        }
      }
    >
      What is your policy on refunds?
    </h4>
    <p
      className=""
      style={
        Object {
          "color": "#212529",
        }
      }
    >
      At this time, we are unable to honor refunds.
    </p>
  </div>
</div>
`;
